### PR TITLE
raftstore/coprocessor: use approximate way to split when meet large region

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
- use approximate way to split when meet large region
- adjust structure to make observer can decide use which policy to split

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/raftstore/coprocessor/split_check/size.rs [hunk 1]
```diff
@@ -143,11 +165,16 @@ impl<C: Sender<Msg> + Send> SplitCheckObserver for SizeCheckObserver<C> {
                 region_size,
                 self.region_max_size
             );
+            // when meet large region use approximate way to produce split keys
+            if region_size >= LARGE_REGION_SIZE {
```
Comment (huachaohuang) [inline src/raftstore/coprocessor/split_check/size.rs]: Should calculate `LARGE_REGION_SIZE` according to region split size.

**Comment by huachaohuang** [comment]
Why remove the tests?

**Comment by Connor1996** [comment]
@huachaohuang before there are two tests `test_split_check()` and `test_batch_split_check()`, and the difference is that `batch_split_limit` is 1 for former and is 5 for latter. Actually, the latter covers the former. And now since when region_size > region_max_size * batch_split_limit * 2, it will use approximate way which produces different split key and make test failed.

**Pull request merged**